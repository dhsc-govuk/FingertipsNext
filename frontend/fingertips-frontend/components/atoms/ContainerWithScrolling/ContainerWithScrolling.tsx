import styled from 'styled-components';

interface ScrollProps {
  horizontal?: boolean;
  vertical?: boolean;
}

export const ContainerWithScrolling = styled.div<ScrollProps>(
  ({ horizontal, vertical }) => ({
    overflowX: horizontal ? 'auto' : 'hidden',
    overflowY: vertical ? 'auto' : 'hidden',
  })
);
