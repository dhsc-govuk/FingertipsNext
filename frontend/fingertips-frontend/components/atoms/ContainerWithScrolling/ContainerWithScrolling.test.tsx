import React from 'react';
import { render } from '@testing-library/react';
import { ContainerWithScrolling } from './ContainerWithScrolling'; // adjust import path

describe('ContainerWithScrolling', () => {
  it('applies both horizontal and vertical scrolling when both props are true', () => {
    const { container } = render(
      <ContainerWithScrolling horizontal vertical />
    );
    const div = container.firstChild as HTMLElement;
    expect(div).toHaveStyle('overflow-x: auto');
    expect(div).toHaveStyle('overflow-y: auto');
  });

  it('applies only horizontal scrolling when only horizontal is true', () => {
    const { container } = render(<ContainerWithScrolling horizontal />);
    const div = container.firstChild as HTMLElement;
    expect(div).toHaveStyle('overflow-x: auto');
    expect(div).toHaveStyle('overflow-y: hidden');
  });

  it('applies only vertical scrolling when only vertical is true', () => {
    const { container } = render(<ContainerWithScrolling vertical />);
    const div = container.firstChild as HTMLElement;
    expect(div).toHaveStyle('overflow-x: hidden');
    expect(div).toHaveStyle('overflow-y: auto');
  });

  it('applies no scrolling when neither prop is provided', () => {
    const { container } = render(<ContainerWithScrolling />);
    const div = container.firstChild as HTMLElement;
    expect(div).toHaveStyle('overflow-x: hidden');
    expect(div).toHaveStyle('overflow-y: hidden');
  });
});
