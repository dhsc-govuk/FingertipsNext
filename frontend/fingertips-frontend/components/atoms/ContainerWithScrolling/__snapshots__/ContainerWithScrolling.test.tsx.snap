// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerWithOutline should render correctly 1`] = `
.c0 {
  overflow-x: hidden;
  overflow-y: hidden;
}

<div>
  <div
    class="c0"
  >
    Content
  </div>
</div>
`;
